services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/RocksDbKeyValueStorage.java

```diff
@@ -154,6 +156,41 @@ public Transaction startTransaction() throws StorageException {
     return new RocksDbTransaction(db.beginTransaction(options), options);
   }
 
+  @Override
+  public long removeUnless(final Predicate<BytesValue> inUseCheck) throws StorageException {
+    long removedNodeCounter = 0;
+    try (final RocksIterator rocksIterator = db.newIterator()) {
+      rocksIterator.seekToFirst();
+      while (rocksIterator.isValid()) {
+        final byte[] key = rocksIterator.key();
+        if (!inUseCheck.test(BytesValue.wrap(key))) {
+          removedNodeCounter++;
+          db.delete(key);
+        }
+        rocksIterator.next();
+      }
+    } catch (final RocksDBException e) {
+      throw new StorageException(e);
+    }
+    return removedNodeCounter;
+  }
+
+  @Override
+  public void clear() {
+    try (final RocksIterator rocksIterator = db.newIterator()) {
+      rocksIterator.seekToFirst();
+      if (rocksIterator.isValid()) {
```

Comment:
I think a guard blocks would look cleaner here, instead of a nested check.

```
      if (!rocksIterator.isValid()) {
        return;
      }
      final byte[] firstKey = rocksIterator.key();
      rocksIterator.seekToLast();
      if (!rocksIterator.isValid()) {
        return;
      }
      db.deleteRange(firstKey, rocksIterator.key());
```